lib/scion.js

```diff
@@ -815,10 +815,26 @@
               return actionRef.call(this._scriptingContext, currentEvent);     //SCXML system variables
             } catch (e){
               var err = {
-                tagname: actionRef.tagname, 
-                line: actionRef.line, 
-                column: actionRef.column,
-                reason: e.message
+                reason: e.message,
+                event: currentEvent
+              };
+              if (typeof actionRef == 'function') {
+                var fnStr = actionRef.toString();
+                var m = fnStr.match(/line_([0-9]+)_column_([0-9]+)\(/);
+                if (m) {
+                  // SCXML
+                  err.line = parseInt(m[1]);
+                  err.column = parseInt(m[2]);
+                } else {
+                  // default
+                  err.actionRef = fnStr;
+                }
+              } else {
+                // I don't know when this can happen since actionRef.call() is always called before
+                // Kept it for backward compatibility but could/should remove it
+                err.tagname = actionRef.tagname;
```

Comment:
The newer version of the engine provides this metadata, so change the code to prefer using this over the penalty of hackily deriving it from the function name.